compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1318,13 +1318,44 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                         && let hir::IsAsync::NotAsync = self.tcx.asyncness(lifetime_ref.hir_id.owner.def_id)
                         && !self.tcx.features().anonymous_lifetime_in_impl_trait
                     {
-                        rustc_session::parse::feature_err(
-                            &self.tcx.sess.parse_sess,
-                            sym::anonymous_lifetime_in_impl_trait,
-                            lifetime_ref.span,
-                            "anonymous lifetimes in `impl Trait` are unstable",
-                        ).emit();
-                        return;
+
+                        match self.tcx.hir().get_generics(lifetime_ref.hir_id.owner.def_id) {
+                            Some(generics) => {
+                                for i in 0..generics.params.len()  {
+
+                                    if !generics.span.contains(generics.params[i].span)  {
+
+                                        let mut diag =  rustc_session::parse::feature_err(
+                                            &self.tcx.sess.parse_sess,
+                                            sym::anonymous_lifetime_in_impl_trait,
+                                            lifetime_ref.span,
+                                            "anonymous lifetimes in `impl Trait` are unstable",
+                                        );
+
+                                        diag.span_label(lifetime_ref.span, "expected named lifetime parameter");
+
+                                        diag.multipart_suggestion("consider introducing a named lifetime parameter",
+                                                                  vec![
+                                                                    (lifetime_ref.span.shrink_to_hi(), "'a ".to_owned()),
+                                                                    (generics.span, "<'a>".to_owned())
```

Comment:
`<'a>` is only correct if there are no explicit generics. When there are explicit generics, you need to insert a `'a, ` just before the first explicit generic parameter.

Reply:
`span_for_param_suggestion` gives a span after the first generic. Can I use that and put `, 'a` instead?